tools/packager/Makefile

```diff
@@ -58,191 +56,129 @@ rpm: el
-		--after-install marathon.systemd.postinst \
-		$(FPM_OPTS_DEB) $(FPM_OPTS) .
-
-
-.PHONY: osx
-osx: toor/osx/$(PREFIX)/bin/marathon
-	fpm -C toor/osx --iteration $(PKG_REL) $(FPM_OPTS_OSX) $(FPM_OPTS) .
+		$(FPM_OPTS_DEB_SYSTEMD) $(FPM_OPTS) .
+
+.PHONY: ubuntu-bionic-1804
+ubuntu-bionic-1804: toor/ubuntu-bionic/lib/systemd/system/marathon.service
+ubuntu-bionic-1804: toor/ubuntu-bionic/$(PREFIX)/share/marathon
+ubuntu-bionic-1804: toor/ubuntu-bionic/etc/default/marathon
+ubuntu-bionic-1804: bin/fpm-docker
+	$(FPM) -C toor/ubuntu-xenial \
+		--iteration $(PKG_REL).ubuntu1804 \
+		$(FPM_OPTS_DEB_SYSTEMD) $(FPM_OPTS) .
+
+.PHONY: debian-jessie-8
+debian-jessie-8: toor/debian-jessie-8/lib/systemd/system/marathon.service
+debian-jessie-8: toor/debian-jessie-8/$(PREFIX)/share/marathon
+debian-jessie-8: toor/debian-jessie-8/etc/default/marathon
+debian-jessie-8: bin/fpm-docker 
+	$(FPM) -C toor/debian-jessie-8 \
+		--iteration $(PKG_REL).debian8 \
+		$(FPM_OPTS_DEB_SYSTEMD) $(FPM_OPTS) .
+
+.PHONY: debian-stretch-9
+debian-stretch-9: toor/debian-stretch-9/lib/systemd/system/marathon.service
+debian-stretch-9: toor/debian-stretch-9/$(PREFIX)/share/marathon
+debian-stretch-9: toor/debian-stretch-9/etc/default/marathon
+debian-stretch-9: bin/fpm-docker
+	$(FPM) -C toor/debian-stretch-9 \
+		--iteration $(PKG_REL).debian9 \
+		$(FPM_OPTS_DEB_SYSTEMD) $(FPM_OPTS) .
+
+toor/%/etc/default/marathon:
+	mkdir -p "$(dir $@)"
+	cp marathon.defaults $@
 
-toor/%/etc/init/marathon.conf: marathon.conf
+toor/el6/etc/init.d/marathon: scripts/el6.init
 	mkdir -p "$(dir $@)"
-	cp marathon.conf "$@"
+	cp scripts/el6.init "$@"
+	chmod 0755 "$@"
 
-toor/%/etc/init.d/marathon: marathon.init
+toor/ubuntu-trusty/etc/init.d/marathon: scripts/ubuntu1404.init
 	mkdir -p "$(dir $@)"
-	cp marathon.init "$@"
+	cp scripts/ubuntu1404.init "$@"
+	chmod 0755 "$@"
 
-toor/%/usr/lib/systemd/system/marathon.service: marathon.service
+toor/%/usr/lib/systemd/system/marathon.service: scripts/marathon.systemd.service
 	mkdir -p "$(dir $@)"
-	cp marathon.service "$@"
+	cp scripts/marathon.systemd.service "$@"
 
-toor/%/lib/systemd/system/marathon.service: marathon.service
+toor/%/lib/systemd/system/marathon.service: scripts/marathon.systemd.service
 	mkdir -p "$(dir $@)"
-	cp marathon.service "$@"
+	cp scripts/marathon.systemd.service "$@"
 
-toor/%/bin/marathon: marathon-runnable.jar
+toor/%/share/marathon: ../../target/universal/marathon-$(PKG_VER)-$(PKG_COMMIT).tgz
+	rm -rf "$(dir $@)"
 	mkdir -p "$(dir $@)"
-	cp marathon-runnable.jar "$@"
-	chmod 755 "$@"
+	tar xzf ../../target/universal/marathon-$(PKG_VER)-$(PKG_COMMIT).tgz -C "$(dir $@)"
+	mv $@-$(PKG_VER)-$(PKG_COMMIT) $@
+	touch $@
+
+../../target/universal/marathon-$(PKG_VER)-$(PKG_COMMIT).tgz:
+	cd ../../ && sbt universal:packageZipTarball
 
-marathon-runnable.jar:
-	cd marathon && sbt assembly && bin/build-distribution
-	cp marathon/target/$@ $@
+bin/fpm-docker: provision-fpm
+	./provision-fpm
+	chmod +x bin/fpm-docker
```

Comment:
great idea.

I think we should use a flag otherwise it will run every time.

```
bin/.fpm-docker-provisioned:
  PROVISION_ONLY=1 bin/fpm-docker
  touch $@
```

etc.